incubator/hnc/internal/validators/namespace.go

```diff
@@ -137,9 +137,9 @@ func (v *Namespace) cannotDeleteSubnamespace(req *nsRequest) admission.Response
 }
 
 func (v *Namespace) illegalCascadingDeletion(ns *forest.Namespace) admission.Response {
-	// Early exit if the namespace allows cascading deletion.
-	if ns.AllowsCascadingDeletion() {
-		return allow("")
+	if !ns.AllowsCascadingDeletion() && len(ns.ChildNames()) > 0 {
+		msg := fmt.Sprintf("Please set allowCascadingDeletion first in the parent namespace.")
+		return deny(metav1.StatusReasonForbidden, msg)
```

Comment:
Maybe we still keep the early exit `if allowCD` and this one can just be `if len > 0`?

Reply:
I thought we want to have *both* ACD set ob parent and all child namespaces, and that's why we can't early return?

Reply:
updated

Reply:
I think previously we allow `parent without ACD; all children with ACD` but now we just don't allow that since parent may not be aware of the cascading deletion when simply deleting the parent. But as long as `parent with ACD`, we would always allow it? (adding @adrianludwin to this discussion)

Reply:
Yeah I have updated the logic to allow deletion as long as parent has ACD